doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -7,64 +7,179 @@ In this chapter, we introduce advanced commands to modify the way Coq
 parses and prints objects, i.e. the translations between the concrete
 and internal representations of terms and commands.
 
-The main commands to provide custom symbolic notations for terms are
-:cmd:`Notation` and :cmd:`Infix`; they will be described in the
-:ref:`next section <Notations>`. There is also a
-variant of :cmd:`Notation` which does not modify the parser; this provides a
-form of :ref:`abbreviation <Abbreviations>`. It is
-sometimes expected that the same symbolic notation has different meanings in
-different contexts; to achieve this form of overloading, Coq offers a notion
-of :ref:`notation scopes <Scopes>`.
-The main command to provide custom notations for tactics is :cmd:`Tactic Notation`.
+When inputing a term to Coq, it goes through several successive steps.
+
+* :gdef:`lexing`: the input is split in a linear stream of tokens, for
+  instance the string ``"1 + 2 * 2"`` will be split into the sequence
+  of five tokens `'1' '+' '2' '*' '2'`
+
+* :gdef:`parsing`: the previous stream is given a tree-like structure, for
+  instance here
+
+  .. code-block:: text
+     :name: after-parsing
+
+             *
+            / \
+           1   +
+              / \
+             2   2
+
+* notation :gdef:`interpretation`: each syntactic element gets translated to a
+  term, for instance :n:`1` can be interpreted as the natural number
+  :n:`S O` then :n:`2` is interpreted as :n:`S (S O)`, then :n:`2 * 2`
+  as :n:`Nat.mul (S (S O)) (S (S O))` and finally our whole term as
+  :n:`Nat.add (S O) (Nat.mul (S (S O)) (S (S O)))`.
+
+Each step is performed independently and there isn't any kind of
+backtracking from any step to a previous one. Then, the result goes
+through the remaining of the proof assistant, that is the elaboration
+and type checking phases discussed everywhere else in this manual. No
```

Comment:
Alas, the doc doesn't have a good description of elaboration or type checking.  Or maybe fair to say "any description".

Reply:
Well, "not any description" is a bit unfair, since we have at least the typing rules. Anyway, that's unrelated to the current PR.

Reply:
The typing rules as presented are not understandable to those who are not already very familiar with proof logic.  There's no explanation of the notation they're written in, really no explanation of how to compute the type of an expression or do typechecking and no examples.

Not something to fix in this PR, but "type checking phases discussed elsewhere" is misleading here given the current state of the doc.